src/control/server/storage/bdev.go

```diff
@@ -79,7 +120,7 @@ type (
 	SmdDevice struct {
 		UUID       string      `json:"uuid"`
 		TargetIDs  []int32     `hash:"set" json:"tgt_ids"`
-		State      string      `json:"state"`
+		State      SmdState    `json:"-"`
```

Comment:
Is this used to return device state from io engine on 'list device'? Please be aware that the device state returned by list device isn't the 'SMD device state', it is 'BiO device state'.

The SMD device state is the state of the device itself, it is persistent and has only two state: FAULTY or NORMAL, but the 'BIO device state' is kind of transient state in DAOS system, it has four types: NORMAL, EVICTED, UNPLUGGED and NEW.

I think that's why you observed 'unknown' state after set a device as FAULTY? (when a device is set as FAULTY, it'll be automatically evicted, and the state showed by list device will be EVICTED).